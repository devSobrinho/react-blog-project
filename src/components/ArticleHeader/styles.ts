import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};
    padding-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    & ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      /* color: ${theme.colors.darkText}; */
      font-size: ${theme.font.sizes.huge};
    }

    @media${theme.media.lteMedium} {
      & ${HeadingStyles} {
        font-size: ${theme.font.sizes.xlarge};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: 2.4rem;
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
