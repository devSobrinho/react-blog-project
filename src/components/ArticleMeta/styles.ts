import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    p {
      font-size: ${theme.font.sizes.small};
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    @media${theme.media.lteMedium} {
      p {
        font-size: ${theme.font.sizes.small};
      }
    }
  `}
`;
