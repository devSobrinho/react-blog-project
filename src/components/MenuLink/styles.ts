import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    color: ${theme.colors.white};
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    font-size: 1.8rem;
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all, 0.3s ease-in-out;

    & :hover {
      color: ${theme.colors.secondary};
      border-right: 0.5rem solid ${theme.colors.secondary};
    }
  `}
`;
