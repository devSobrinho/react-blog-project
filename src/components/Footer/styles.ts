import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
