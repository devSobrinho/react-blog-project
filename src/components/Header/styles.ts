import styled, { css } from 'styled-components';
import { Title as HeaddingStyles } from '../Heading/styles';

export const Content = styled.div`
  ${({ theme }) => css`
    max-width: 48rem;
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      margin-left: 0;

      & > h2 {
        text-align: center;
      }

  `}
`;
export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    padding-bottom: ${theme.spacings.xhuge};
    margin: 0 auto;

    & ${HeaddingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.lteSmall} {
      display: flex;
      flex-flow: column wrap;
      & > ${HeaddingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }
    }
  `}
`;
