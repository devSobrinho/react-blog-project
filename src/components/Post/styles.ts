import styled, { css } from 'styled-components';
import { Container as HtmlContentStyles } from '../HtmlContent/styles';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.sizes.max};

    & ${HtmlContentStyles} {
      margin-top: ${theme.spacings.xhuge};
    }
  `}
`;
