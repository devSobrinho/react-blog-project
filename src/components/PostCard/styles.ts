import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;

    ${HeadingStyles} {
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: calc(${theme.spacings.small} - 0.6rem);
    }

    ${HeadingStyles}:hover {
      color: ${theme.colors.primary};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    &:hover ${Cover} {
      opacity: 0.8;
    }
    &:hover a {
      color: ${theme.colors.secondary};
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: all 300ms ease-in-out;
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
