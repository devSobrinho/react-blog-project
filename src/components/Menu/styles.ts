import styled, { css } from 'styled-components';
import { Container as LogoLinkStyles } from '../LogoLink/styles';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as MenuLinkStyles } from '../MenuLink/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

export const Wrapper = styled.menu`
  ${() => css`
    position: fixed;
    top: 0;
    z-index: 1;
  `}
`;

const menuVisibleMobile = () => css`
  padding: 3px;
  margin-left: calc(100vw - 39px - 1.6rem);
`;
const menuVisibleDesk = () => css`
  margin-left: 14rem;
`;

export const MenuOpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    border-radius: 5px;
    border-color: transparent;
    padding: 5px;
    margin-left: 1.6rem;
    margin-top: 1.6rem;
    transition: all 300ms ease-in-out;
    &:hover {
      color: ${theme.colors.secondary};
    }

    ${menuVisible && menuVisibleDesk()}

    @media ${theme.media.lteSmall} {
      ${menuVisible && menuVisibleMobile()}
    }
  `}
`;

export const Container = styled.nav<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    width: ${menuVisible ? '20rem' : '1rem'};
    transition: all 300ms ease-in-out;
    height: 100vh;
    background: ${theme.colors.primary};
    padding: ${menuVisible ? theme.spacings.medium : '0.5rem'};
    overflow-y: auto;
    overflow: auto;
    overflow-wrap: anywhere;

    & ${HeadingStyles} {
      width: auto;
      margin-top: 5rem;
    }

    & ${LogoLinkStyles} img {
      border: 0.5rem solid ${theme.colors.secondary};
      border-radius: 50%;
    }

    .menu-navbar {
      list-style: none;
      padding: 0;
      margin: 0;
      width: auto;
    }

    .navbar-item ${MenuLinkStyles} {
    }

    @media ${theme.media.lteSmall} {
      width: ${menuVisible ? '100vw' : '0px'};
      padding: ${menuVisible ? theme.spacings.medium : '0rem'};

      .menu-navbar {
        margin: 0 20px;
      }
      .navbar-item ${MenuLinkStyles} {
        width: initial;
        border: none;
      }
    }
  `}
`;
