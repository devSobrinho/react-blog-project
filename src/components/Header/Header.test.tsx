import { screen } from '@testing-library/react';
import { Header } from '.';
import { renderTheme } from '../../styles/render-theme';

import props from './mock';

describe('<Header />', () => {
  it('should render an image, a heading and text', () => {
    renderTheme(<Header {...props} showText={undefined} />);

    expect(
      screen.getByRole('heading', { name: props.blogName }),
    ).toBeInTheDocument();
    expect(screen.getByText(`${props.blogDescription}`)).toBeInTheDocument();
    expect(
      screen.getByAltText(`${props.blogName} - ${props.blogDescription}`),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('img', {
        name: `${props.blogName} - ${props.blogDescription}`,
      }),
    ).toHaveAttribute('src', props.logo.url);
  });

  it('should render image only', () => {
    renderTheme(<Header {...props} showText={false} />);

    expect(
      screen.getByAltText(`${props.blogName} - ${props.blogDescription}`),
    ).toBeInTheDocument();
    expect(
      screen.queryByRole('heading', { name: props.blogName }),
    ).not.toBeInTheDocument();
    expect(screen.queryByText(props.blogDescription)).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Header {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
