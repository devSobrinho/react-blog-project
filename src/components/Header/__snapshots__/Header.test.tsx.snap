// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
.c2 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c5 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c3 > img {
  height: 15rem;
  width: 15rem;
  border-radius: 50%;
}

.c4 {
  max-width: 48rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column wrap;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  margin-left: 3.2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  padding-bottom: 6.4rem;
  margin: 0 auto;
}

.c0 .c1 {
  margin: 0 0 calc(1.6rem - 1rem);
}

<header
  class="c0"
>
  <h1
    class="c1 c2"
  >
    <a
      class="c3"
      href="/"
      target="_self"
    >
      <img
        alt="Daniel Sobrinho - Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud."
        src="assets/images/dev-sobrinho.jpeg"
      />
    </a>
  </h1>
  <div
    class="c4"
  >
    <h2
      class="c1 c5"
    >
      Daniel Sobrinho
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.
    </p>
  </div>
</header>
`;
